### ISSUE TITLE
Street Fighter Alpha 2: one does not work, one glitchy

### ISSUE THREAD

**Issue by codex785** [ClusterM/hakchi2#492]
i have 2 versions of SF alpha 2 sfrom
one glitchy, one not working and C7 on press reset.
@ cluster do you want to have a look at it?

**Comment by ghost** [comment]
They used a hacked rom for the virtual console release with decompressed graphics inside of it since they don't emulate the S-DD1 chip at all. You'd have to extract said rom from it and use that instead or just play it using retroarch

**Comment by ghost** [comment]
@ClusterM do you think you could take a look at this? I made a bps patch to turn the regular street fighter alpha 2 (U) dump into the virtual console one https://mega.nz/#!YuxRBAwL!KLrxupkrpLMWud8Suz4M7MixBdcw0RWAUdNfb04bz_Y

**Comment by ClusterM** [comment]
@spring568 so, is it working?

**Comment by ghost** [comment]
Someone else would just need to upload the rom with out converting it to see if it works since I don't have a mini snes.

**Comment by GerryHolt** [comment]
After applying the patch provided by spring, the game sits at a black screen until I hit the reset button, then I get error C7

**Comment by isage** [comment]
If someone can provide original VC SFA2 rom (data.bin), then we can figure out how decompressed sdd-1 data is stored (it should be appended at the end of sfrom and 0x20 field in header should point to it's address, but the actual format is unknown) and write a converter.

**Comment by ghost** [comment]
My patch does turn it into the original data.bin. The thing is it doesn't have a snes vc header at the bottom too so that would need to be added using a hex editor. That's why I was hoping someone more experienced with this could look at it. If it helps the crc32 of the rom to patch is 9C59DDFF

Edit: I'm sure if it was uploaded using hakchi2 it would just add a second VC header at the top thus not allowing it to boot at all

**Comment by isage** [comment]
On 3ds second header is right after first one, and it doesn't really matter
where it is, as long as first one points to right location. I'll play with
your patch later. Maybe it needs correct game preset flag, or, there's sad
possibility, that snesmini emu doesn't support unpacke sdd1 at all

**Comment by ghost** [comment]
Well there's no unpacking at all. Like I said it's a hack I think similar to the star ocean one. You can see right after the end of the regular rom dump if you remove the snes vc header at the top then compare it to the one from a real snes. The locations that tell it how to decompress graphics are replaced with something at the beginning too that say SDD1 in ascii. Basically this snes vc data.bin wont work on a real super nintendo only their emulator, It's a 10 meg file. If you trim the rom to the proper size of 4MB all the graphics in game will show up as corrupt in snes9x or a real snes because it's a hack

**Comment by isage** [comment]
it's the same as with brr->pcm. sdd1 data is replaced with magick (sdd1 in
ascii) + pointer to unpacked data (at the end of data.bin).

On Oct 8, 2017 2:07 PM, "spring568" <notifications@github.com> wrote:

Well there's no unpacking at all. Like I said it's a hack I think similar
to the star ocean one. You can see right after the end of the regular rom
dump if you remove the snes vc header at the top then compare it to the one
from a real snes. The locations that tell it how to decompress graphics are
replaced with something at the beginning too that say SDD1 in ascii.
Basically this snes vc data.bin wont work on a real super nintendo only
their emulator

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/ClusterM/hakchi2/issues/492#issuecomment-334999061>,
or mute
the thread

**Comment by ghost** [comment]
Ah okay, thanks @isage. But yeah I was hoping you or someone else could figure out some magik to get it to load ;). Does the emulator have some kind of patching system at all? I saw someone extract some strings that mentioned that for the snes classic. Wonder if it needs another file to help it load.. or capcom put in a kludge in only that version of the emulator which would make it impossible to load.

**Comment by ghost** [comment]
Edit: hmm https://www.3dbrew.org/wiki/3DS_Virtual_Console#RomFS_2 Well there's my answer. It looks like dark akuma might have gotten it working too with further research, but no idea. Once the format is finalized I can make a new bps patch for the rom. It's a lot easier then using an IPS patch

**Comment by Dark-Akuma** [comment]
I did figure SFA2 support out. ATM I'm waiting on contact back from cluster to discuss some changes to hakchi2 so it can be supported properly.

**Comment by Darky2k1** [comment]
@Dark-Akuma Mind just sharing the details with the group? Since Hakchi is open source any of us can put in a pull request to make the necessary changes....

**Comment by isage** [comment]
And not only pull-requests. Hakchi2 is windows-only, so linux/osx users would need some different converter.
Also, @Dark-Akuma what about star ocean?

**Comment by Dark-Akuma** [comment]
@darkunderdog Just posted my details here. 

https://www.reddit.com/r/miniSNES/comments/75ffcj/street_fighter_alpha_2_support_in_canoe_is/

@isage My conclusions about Star Ocean are at the bottom. In short, no. Most likely won't ever work with canoe. =(

**Comment by Drakonas** [comment]
This should have not ever been posted on this tracker. This is a Nintendo code issue, not Hakchi. Rom compatibility discussion with Canoe should be kept on Discord or Reddit or elsewhere. This is for reporting bugs with Hakchi, not canoe. Closing.
